### PR TITLE
Rework to clarify supported scenarios

### LINKED ISSUE #37142: Important NuGet packages are not named or linked to
This document mentions that we can use NuGet packages to enable support for newer language features on older frameworks. It should also give information about _which_ NuGet packages these are, or how to find them.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 7d0be2e3-8f90-8843-f98c-aadba19bec6d
* Version Independent ID: 012620d5-fa79-cc15-341b-3da8e5748890
* Content: [The relationship between language features and library types - C#](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/relationships-between-language-and-library)
* Content Source: [docs/csharp/whats-new/relationships-between-language-and-library.md](https://github.com/dotnet/docs/blob/main/docs/csharp/whats-new/relationships-between-language-and-library.md)
* Product: **dotnet-csharp**
* Technology: **csharp-whats-new**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

---
[Associated WorkItem - 161150](https://dev.azure.com/msft-skilling/Content/_workitems/edit/161150)

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #37142

Using NuGet packages to monkey patch missing features in older frameworks is possible, but not supported. Our docs shouldn't encourage it.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/whats-new/relationships-between-language-and-library.md](https://github.com/dotnet/docs/blob/236149a9c48e77e95d00aa75196b532a45a50ddb/docs/csharp/whats-new/relationships-between-language-and-library.md) | [Relationships between language features and library types](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/relationships-between-language-and-library?branch=pr-en-us-37736) |

### CODE REVIEW

File: docs/csharp/whats-new/relationships-between-language-and-library.md [hunk 1]
```diff
@@ -1,55 +1,21 @@
...
 
 # Relationships between language features and library types
 
-The C# language definition requires a standard library to have certain
-types and certain accessible members on those types. The compiler generates
-code that uses these required types and members for many different language
-features. When necessary, there are NuGet packages that contain types
-needed for newer versions of the language when writing code for environments
-where those types or members have not been deployed yet.
+The C# language definition requires a standard library to have certain types and certain accessible members on those types. The compiler generates code that uses these required types and members for many different language features. For this reason, C# versions are supported only for the corresponding .NET version and newer. That ensures the correct runtime behavior and the availability of all required types and members.
```
Comment (gewarren) [inline docs/csharp/whats-new/relationships-between-language-and-library.md:9]:
```suggestion
The C# language definition requires a standard library to have certain types and certain accessible members on those types. The compiler generates code that uses these required types and members for many different language features. For this reason, C# versions are supported only for the corresponding .NET version and newer. That ensures the correct run-time behavior and the availability of all required types and members.
```

File: docs/csharp/whats-new/relationships-between-language-and-library.md [hunk 2]
```diff
@@ -1,55 +1,21 @@
...
 
-This dependency on standard library functionality has been part of the
-C# language since its first version. In that version, examples included:
+This dependency on standard library functionality has been part of the C# language since its first version. In that version, examples included:
 
-* <xref:System.Exception> - used for all compiler generated exceptions.
-* <xref:System.String> - the C# `string` type is a synonym for <xref:System.String>.
-* <xref:System.Int32> - synonym of `int`.
+- <xref:System.Exception> - used for all compiler generated exceptions.
```
Comment (gewarren) [inline docs/csharp/whats-new/relationships-between-language-and-library.md:13]:
```suggestion
- <xref:System.Exception> - used for all compiler-generated exceptions.
```

File: docs/csharp/whats-new/relationships-between-language-and-library.md [hunk 3]
```diff
@@ -1,55 +1,21 @@
...
+- <xref:System.String> - the C# `string` type is a synonym for <xref:System.String>.
```
Comment (gewarren) [inline docs/csharp/whats-new/relationships-between-language-and-library.md:14]:
```suggestion
- <xref:System.String> - synonym of `string`.
```

Comment (gewarren) [reply docs/csharp/whats-new/relationships-between-language-and-library.md:14]: To match the next line...

File: docs/csharp/whats-new/relationships-between-language-and-library.md [hunk 4]
```diff
@@ -1,55 +1,21 @@
...
-the dependencies. Examples include: <xref:System.Runtime.CompilerServices.INotifyCompletion>,
-<xref:System.Runtime.CompilerServices.CallerFilePathAttribute> and
-<xref:System.Runtime.CompilerServices.CallerMemberNameAttribute>. C# 7.0 continues this by adding a dependency on <xref:System.ValueTuple> to
-implement the [tuples](../language-reference/builtin-types/value-tuples.md) language feature.
+Subsequent versions of C# have occasionally added new types or members to the dependencies. Examples include: <xref:System.Runtime.CompilerServices.INotifyCompletion>, <xref:System.Runtime.CompilerServices.CallerFilePathAttribute> and <xref:System.Runtime.CompilerServices.CallerMemberNameAttribute>. C# 7.0 added a dependency on <xref:System.ValueTuple> to implement the [tuples](../language-reference/builtin-types/value-tuples.md) language feature. C# 8 requires <xref:System.Index?displayProperty=nameWithType> and <xref:System.Range?displayProperty=nameWithType> for [ranges and indexes](../language-reference/operators/member-access-operators.md#range-operator-), among other features. Each new version might add additional requirements.
```
Comment (gewarren) [inline docs/csharp/whats-new/relationships-between-language-and-library.md:19]:
```suggestion
Subsequent versions of C# have occasionally added new types or members to the dependencies. Examples include: <xref:System.Runtime.CompilerServices.INotifyCompletion>, <xref:System.Runtime.CompilerServices.CallerFilePathAttribute>, and <xref:System.Runtime.CompilerServices.CallerMemberNameAttribute>. C# 7.0 added a dependency on <xref:System.ValueTuple> to implement the [tuples](../language-reference/builtin-types/value-tuples.md) language feature. C# 8 requires <xref:System.Index?displayProperty=nameWithType> and <xref:System.Range?displayProperty=nameWithType> for [ranges and indexes](../language-reference/operators/member-access-operators.md#range-operator-), among other features. Each new version might add additional requirements.
```

**Review by gewarren — APPROVED** [review]
Looks good. Left a couple suggestions.

File: docs/csharp/whats-new/relationships-between-language-and-library.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/whats-new/relationships-between-language-and-library.md:9]: that's what I thought, but Acrolinx complains about it.

**Pull request merged**